spikeextractors/multirecordingtimeextractor.py

```diff
@@ -26,14 +28,26 @@ def __init__(self, recordings, epoch_names=None):
             start_frames += num_frames
 
         # Test if all recording extractors have same num channels and sampling frequency
+        reset_channels = False
         for i, recording in enumerate(recordings[1:]):
             channel_ids = recording.get_channel_ids()
             sampling_frequency = recording.get_sampling_frequency()
 
-            if (self._channel_ids != channel_ids):
-                raise ValueError("Inconsistent channel ids between extractor 0 and extractor " + str(i + 1))
-            if (self._sampling_frequency != sampling_frequency):
-                raise ValueError("Inconsistent sampling frequency between extractor 0 and extractor " + str(i + 1))
+            assert len(self._channel_ids) == len(channel_ids), f"Inconsistent number of channels between " \
+                                                               f"extractor 0 and extractor {i + 1}"
+            assert self._sampling_frequency == sampling_frequency, f"Inconsistent sampling frequency between " \
+                                                                   f"extractor 0 and extractor {i + 1}"
+            assert np.array_equal(self._first_recording.get_channel_locations(), recording.get_channel_locations()), \
+                f"Inconsistent locations between extractor 0 and extractor {i + 1}"
+            if self._channel_ids != channel_ids:
```

Comment:
This looks like very very permissive. What is the suer case ?

Reply:
We use several channel configuration of 1000 channels but only 100-200 shared channels covering hot spots. I need to sort those 200 channels and concatenate them

Reply:
You should concatenate 2 subrecording with the aligned channel_ids so. No ?
Making hidden magic is not a good practice, I think.

Reply:
No. They have different chanel ids because they have different configurations, but shared channels with the same location

Reply:
I have an internal warning flag that is flashing in red and telling me that must be done in two steps with 2 distinct functions(classes) : 1. align/select channel with same localtion 2. concatenate in time.
I know your tenacity : you will convice me but at the moment it is not the case.

Reply:
of course I do that externally! I think the use case is so niche that it shouldn't make it into the class...